map_redcap_to_flywheel_service/utils/yaml_decoder_rc2fw.py

```diff
@@ -117,40 +118,126 @@ def get_redcap_mc_map2(project):
     return(redcap_reassign_map)
 
 
+def remove_repeat_instances(records, record_id, foids):
+    ''' Remover, via merging, any repeat records.
+        Each subject with N repeating instances will appear to have N records.
+        The records for each subject can contain partial information, thus the
+        need for merging and not simply locating the non-empty record
+
+        Args:
+            records (list(dict)): list of dictionary objects containing REDCap
+                                  subject information
+            record_id (str): key to obtain subjects ID from data dictionary
+            foids (list(str)): list of fields of interest (field names from
+                               REDCap)
+
+        Returns:
+            records (list(dict)): list of dict objects containing one
+                                  dict of complete information per subject
+    '''
+
+    # Create a dict recording how many repeat entries for each subject
+    labels = {}
+    for r in records:
+        id = r[record_id]
+        if id in labels.keys():
+            labels[id] += 1
+        else:
+            labels[id] = 1
+
+    # Tally how many subjects have multiple entries
+    res = Counter(labels.values())
+
+    # Check if any subjects were retrieved multiple times
+    if max(res.keys()) == 1:
+        log.info("No duplicates found")
+        return records
+
+    log.info("Subjects with repeated instruments found")
+    log.info("Distribution of multiple records (num records : num subjects)")
+    log.info(res)
+
+    # Loop over each subject and create new data entry from the multiple entries
+    for id, entries in labels.items():
+        if entries == 1:
+            continue
+
+        # Select out multiple records
+        temp_data = [r for r in records if r[record_id] == id]
+
+        # Remove that data from original dict
+        records = [r for r in records if r[record_id] != id]
+
+        # Loop over FOI's and create new dict entry for subject
+        new_dict = {}
+        for i in foids:
+            new_dict[i] = None
+            new_value = ''
+            old_value = ''
+            for j, entry in enumerate(temp_data):
+                if j == 0:
```

Comment:
I think this is redundant since we're initializing to '' just before entering the loop

Reply:
Yes, that is redundant. Sorry! Removed in 34b67df